lib/normalize_grammar.re

```diff
@@ -1,5 +1,78 @@
 open Common;
 
-let normalize = _ast =>
-  raise (Todo);
-  
\ No newline at end of file
+let normalize_to_simple = body => {
+  switch (body) {
+  | Ast_grammar.TOKEN =>
+    Ast_grammar_normalized.ATOM(
+      Ast_grammar_normalized.TOKEN
+    );
+  | Ast_grammar.SYMBOL(name)=>
+    Ast_grammar_normalized.ATOM(
+      Ast_grammar_normalized.SYMBOL(name)
+    );
+  /* Ignore rest */
+  | _ => {
+    print_string("Inside normalize_to_simple\n");
+    raise(Todo);
+  }
+  }
+}
+
+let normalize_to_atom = body => {
+  switch (body) {
+  | Ast_grammar.TOKEN =>
+    Ast_grammar_normalized.TOKEN
+  | Ast_grammar.SYMBOL(name)=>
+    Ast_grammar_normalized.SYMBOL(name)
+  /* Ignore rest */
+  | _ => {
```

Comment:
in this case, you'll need to generate a fresh identifier, to introduce a new SYMBOL with this
body to be processed later on by normalize_to_rule_body.

| _ => {
   let fresh_ident = gensym();
   B.SYMBOL(fresh_ident), [fresh_ident, body]

Reply:
Ok.